bin/verify-era-proof-attestation/src/main.rs

```diff
@@ -1,221 +1,91 @@
-    let mut process_handle = tokio::spawn(verify_batches_proofs(stop_receiver, args));
-    tokio::select! {
-        ret = &mut process_handle => { return ret?; },
-        _ = signal::ctrl_c() => {
-            tracing::info!("Stop signal received, shutting down");
-            stop_sender.send(true).ok();
-            // Wait for process_batches to complete gracefully
-            process_handle.await??;
-        }
-    }
-
-    Ok(())
-}
-
-fn validate_arguments(args: &Arguments) -> Result<()> {
-    if args.attestation_policy.sgx_mrsigners.is_none()
-        && args.attestation_policy.sgx_mrenclaves.is_none()
-    {
-        error!("Neither `--sgx-mrenclaves` nor `--sgx-mrsigners` specified. Any code could have produced the proof.");
-    }
-
-    Ok(())
-}
-
-/// Verify all TEE proofs for all batches starting from the given batch number up to the specified
-/// batch number, if a range is provided. Otherwise, continue verifying batches until the stop
-/// signal is received.
-async fn verify_batches_proofs(
-    mut stop_receiver: watch::Receiver<bool>,
-    args: Arguments,
-) -> Result<()> {
-    let node_client = MainNodeClient::new(args.rpc_url.clone(), args.chain_id)?;
-    let http_client = Client::new();
-    let first_batch_number = match args.batch_range {
-        Some((first_batch_number, _)) => first_batch_number,
-        None => args
-            .continuous
-            .expect("clap::ArgGroup should guarantee batch range or continuous option is set"),
-    };
-    let end_batch_number = args
-        .batch_range
-        .map_or(u32::MAX, |(_, end_batch_number)| end_batch_number.0);
-    let mut unverified_batches_count: u32 = 0;
-    let mut last_processed_batch_number = first_batch_number.0;
-
-    for current_batch_number in first_batch_number.0..=end_batch_number {
-        if *stop_receiver.borrow() {
-            tracing::warn!("Stop signal received, shutting down");
-            break;
-        }
+    // Parse command-line arguments
+    let config = VerifierConfig::new(VerifierConfigArgs::parse())?;
 
-        trace!("Verifying TEE proofs for batch #{}", current_batch_number);
+    // Initialize logging
+    tracing::subscriber::set_global_default(
+        teepot::log::setup_logging(env!("CARGO_CRATE_NAME"), &config.args.log_level)
+            .map_err(|e| Error::internal(e.to_string()))?,
+    )
+    .map_err(|e| Error::internal(e.to_string()))?;
 
-        let all_verified = verify_batch_proofs(
-            &mut stop_receiver,
-            current_batch_number.into(),
-            &args.rpc_url,
-            &http_client,
-            &node_client,
-            &args.attestation_policy,
-        )
-        .await?;
+    // Create processor based on config
+    let (processor, mode) = ProcessorFactory::create(config.clone())?;
 
-        if !all_verified {
-            unverified_batches_count += 1;
-        }
+    // Set up stop channel
+    let (stop_sender, stop_receiver) = watch::channel(false);
```

Comment:
Stop channel should probably be a cancellation token instead. https://tokio.rs/tokio/topics/shutdown